### PR TITLE
For embeddings, use `sentence_*_config.json`, download HF async, use TEI functions

### PR DESCRIPTION
**Pull request by Jeadie**
## 🗣 Description
Did some refactoring:
 - Read 'sentence_*_config.json' for HF models. 
   - Add `params.max_seq_length` for HF/local models.  
   - Solves this [TODO](https://github.com/spiceai/spiceai/pull/3724/files#diff-0538e7ae41d95b18daf9d247c9ef79c1d04951629500ab9000c142e723c40ae1L141-L142). 
 - Use `tokio` based `hf_hub` to download HF files
 - Use code from [text-embeddings-inference](https://github.com/huggingface/text-embeddings-inference) to download files.

## 📄 Documentation Requirements
 - Add `params.max_seq_length` to embedding components.

## Breaking Changes
 - This will break the following use case of chunked embedding datasets.
    - A dataset with `columns[].embeddings.chunking.target_chunk_size` set
    - Using [`from: huggingface:huggingface.co/sentence-transformers/all-MiniLM-L6-v2`](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2)
    - If `target_chunk_size` > 256, the dataset will error with `Input validation error: `inputs` must have less than 256 tokens. Given: 306` (because of its [sentence_bert_config.json](https://huggingface.co/sentence-transformers/all-MiniLM-L6-v2/blob/main/sentence_bert_config.json)).

### CODE REVIEW

File: crates/llms/Cargo.toml [hunk 1]
```diff
@@ -33,7 +33,11 @@ tracing.workspace = true
 tracing-futures.workspace = true
 
 ## For Chunking
-text-splitter = {  git = "https://github.com/Jeadie/text-splitter.git", rev = "b33e4748576b9a81d436d3f28a6706f860fc45ba", features = ["markdown", "tokenizers", "tiktoken-rs"]}
+text-splitter = { git = "https://github.com/Jeadie/text-splitter.git", rev = "b33e4748576b9a81d436d3f28a6706f860fc45ba", features = [
```
Comment (phillipleblanc) [inline crates/llms/Cargo.toml]:
```suggestion
text-splitter = { git = "https://github.com/spiceai/text-splitter.git", rev = "b33e4748576b9a81d436d3f28a6706f860fc45ba", features = [
```

Comment (phillipleblanc) [reply crates/llms/Cargo.toml]: I've created the fork and pushed commit `b33e4748576b9a81d436d3f28a6706f860fc45ba` to the `spiceai` branch

File: crates/llms/src/embeddings/mod.rs [hunk 2]
```diff
@@ -59,6 +59,9 @@ pub enum Error {
         source: Box<dyn std::error::Error + Send + Sync>,
     },
 
+    #[snafu(display("An issue occurred with Huggingface: {source} "))]
```
Comment (phillipleblanc) [inline crates/llms/src/embeddings/mod.rs]: This error should follow the new error guidelines

File: crates/llms/src/embeddings/candle/tei.rs [hunk 3]
```diff
@@ -127,20 +129,36 @@ impl TeiEmbed {
...
-        // TODO: `max_input_length` should take into account overwrite files like `sentence_bert_config.json`.
-        // See `<https://github.com/huggingface/text-embeddings-inference/blob/cb1e594709fb1caea674ed460b6e426b2b4a531b/router/src/lib.rs#L189>`
-        let max_input_length = config.max_position_embeddings - position_offset;
+
+        let max_input_length = if let Some(max_seq_length) = max_seq_length_overwrite {
+            max_seq_length
+        } else {
+            // Some models will have `sentence_*_config.json` file defining a specific `max_seq_length`.
+            match max_seq_length_from_st_config(root) {
+                Ok(Some(max_seq_length)) => max_seq_length,
```
Comment (sgrebnov) [inline crates/llms/src/embeddings/candle/tei.rs]: The following version is easier to read or understand IMO (assume `_ =>` refers to Ok(None)):
Ok(max_seq_length) => max_seq_length.unwrap_or(config.max_position_embeddings - position_offset)

**Review by sgrebnov — DISMISSED** [review]

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**